pkg/goDB/encoder/lz4/lz4.go

```diff
@@ -155,10 +155,10 @@ func (e *Encoder) Compress(data, buf []byte, dst io.Writer) (n int, err error) {
 	}
 	buf = buf[:dstCapacity]
 
-	var compLen = int(C.lz4Compress(
-		unsafe.Pointer(&data[0]),
+	compLen := int(C.lz4Compress(
+		(*C.char)(unsafe.Pointer(&data[0])),
```

Comment:
Just out of curiosity: does this explicit cast improve performance?

Reply:
Yes and no. As in: It doesn't do anything directly for performance, but casting from / to CGo has some nasty caveats I learned of recently, especially regarding escape analysis / memory allocations. TL;DR: The more explicit types are cast between Go and CGO the better for the compiler). We know that in C this is a char* (a.k.a. byte array), so we can give the compiler that hint. Together with the const type modifiers this removed half of the remaining memory objects allocations (because before, the compiler was allocating a copy for these pointers AND the constant integers for no reason on each call).